cli/azd/cmd/container_test.go

```diff
@@ -0,0 +1,88 @@
+package cmd
+
+import (
+	"context"
+	"testing"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/ioc"
+	"github.com/azure/azure-dev/cli/azd/pkg/lazy"
+	"github.com/azure/azure-dev/cli/azd/pkg/project"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_Lazy_Project_Config_Resolution(t *testing.T) {
+	ctx := context.Background()
+	container := ioc.NewNestedContainer(nil)
+	ioc.RegisterInstance(container, ctx)
+
+	registerCommonDependencies(container)
+
+	// Register the testing lazy component
```

Comment:
YMMV, but if we wanted a more direct test, we could validate `Lazy[*project.ProjectConfig].GetValue() == *project.ProjectConfig` by resolving these instances directly without introducing the additional `testLazyComponent` and `testConcreteComponent`.

Reply:
Yes - that would be more straight forward but wanted the test to resolve via the IoC container that more closely aligns to how components take dependencies on these components.